hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -113,6 +113,45 @@ class LowerTableIR(val typesToLower: DArrayLowering.Type) extends AnyVal {
             throw new LowererUnsupportedOperation(s"can't lower a TableRead with reader $r.")
         }
 
+      case TableParallelize(rowsAndGlobal, nPartitions) =>
+        val nPartitionsAdj = nPartitions.getOrElse(16)
+        val loweredRowsAndGlobal = lowerIR(rowsAndGlobal)
+
+        val contextType = TStruct(
+          "elements" -> TArray(GetField(loweredRowsAndGlobal, "rows").typ.asInstanceOf[TArray].elementType)
+        )
+
+        val context = MakeStream((0 until nPartitionsAdj).map { partIdx =>
+          val numRows = ArrayLen(GetField(loweredRowsAndGlobal, "rows"))
+          val numNonEmptyPartitions = If(numRows < nPartitionsAdj, numRows, nPartitionsAdj)
+          val q = numRows floorDiv numNonEmptyPartitions
+          val remainder = numRows - q * numNonEmptyPartitions
+          val length = If(numNonEmptyPartitions >= partIdx,
+            If(remainder > 0,
+              If(remainder > partIdx, q + 1, q),
+              q),
+            0
+          )
+          val start = If(numNonEmptyPartitions >= partIdx,
```

Comment:
There is a conditional-free closed for expression for this.  See `utils.partition`.

Reply:
That thing computes all the lengths. I saw that, but wasn't am not aware of a closed form expression for the starting positions. I'll change the length code to use the simpler thing though.